specs/hotel_spec.rb

```diff
@@ -0,0 +1,221 @@
+      @hotel.number_of_rooms.must_equal 20
+    end
+
+    it "has a list_of_rooms attribute" do
+      @hotel.must_respond_to :list_of_rooms
+    end
+
+    it "The list_of_rooms attribute that is an array of rooms" do
+      @hotel.list_of_rooms.must_be_kind_of Array
+      @hotel.list_of_rooms[0].must_be_kind_of Hotel::Room
+      @hotel.list_of_rooms[0].number.must_equal 1
+      @hotel.list_of_rooms[19].number.must_equal 20
+      @hotel.list_of_rooms[20].must_be_nil
+      @hotel.list_of_rooms[0].price.must_equal 200.00
+    end
+
+    it "has a make_reservation method" do
+      @hotel.must_respond_to :make_reservation
+
+    end
+
+    it "has a reservations attribute" do
+      @hotel.must_respond_to :reservations
+    end
+
+    it "The reservations attribute is an array of reservations" do
+      @hotel.make_reservation("marcel luedtke", "2017/9/17", "2017/9/19", 1)
+      @hotel.reservations.must_be_kind_of Array
+      @hotel.reservations[0].client.must_equal "Marcel Luedtke"
+      @hotel.reservations[0].stay_length.must_equal 2
+    end
+
+    it "The make_reservation method update the status of the rooms booked" do
+      @hotel = Hotel::Hotel.new(20)
+      @hotel.list_of_rooms[0].booked.must_be_empty
+      @hotel.make_reservation("marcel luedtke", "2017/9/17", "2017/9/19", 1)
+      @hotel.list_of_rooms[0].booked[0].must_be_kind_of Hash
+      @hotel.list_of_rooms[0].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[0].booked[0]["arrival"].year.must_equal 2017
+      @hotel.list_of_rooms[0].booked[0]["arrival"].mon.must_equal 9
+      @hotel.list_of_rooms[0].booked[0]["arrival"].mday.must_equal 17
+      @hotel.list_of_rooms[0].booked[0]["departure"].year.must_equal 2017
+      @hotel.list_of_rooms[0].booked[0]["departure"].mon.must_equal 9
+      @hotel.list_of_rooms[0].booked[0]["departure"].mday.must_equal 19
+      @hotel.list_of_rooms[1].booked.must_be_empty
+      @hotel = Hotel::Hotel.new(20)
+      @hotel.make_reservation("Walter luedtke", "2017/9/17", "2017/9/19", 3)
+      @hotel.list_of_rooms[0].booked[0].must_be_kind_of Hash
+      @hotel.list_of_rooms[0].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[1].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[2].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[3].booked.must_be_empty
+      @hotel.list_of_rooms[4].booked.must_be_empty
+      @hotel.list_of_rooms[5].booked.must_be_empty
+      @hotel.make_reservation("Alex luedtke", "2017/9/17", "2017/9/19", 1)
+      @hotel.list_of_rooms[0].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[1].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[2].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[3].booked[0]["arrival"].class.must_equal Date
+      @hotel.list_of_rooms[4].booked.must_be_empty
+      @hotel.list_of_rooms[5].booked.must_be_empty
```

Comment:
Most of these lines (lines 53-77) are testing for the properties on room.

In the `Hotel` test, we want to test for things that are closely related to hotel, and in the `Room` test, we want to test for things that are closely related to room. It's not bad to test for the properties of a given room in `@hotel.list_of_rooms` but it's a small code smell to need to specify several layers into `room` what its internals are.